Articles/Blog/2023-11-27-julia-bugs.md

```diff
@@ -0,0 +1,37 @@
+# Come for Syntax, Stay for Speed, and Understand Bugs in Julia Programs 
+
+**Hero Image:**
+
+ - <img src='../../images/julia-bugs-taxonomy.png' />
+ 
+
+#### Contributed by [Akond Rahman](https://github.com/akondrahman "Akond Rahman's GitHub Profile")
+
+#### Publication date: November 27, 2023 
+
+In this blog post, I describe some of our [research findings](https://link.springer.com/article/10.1007/s10664-023-10328-5) related to bugs that occur in Julia programs, which are becoming increasing popular to implement scientific software. 
+
+Since its inception in 2012, Julia has experienced a steady increase in popularity as practitioners are migrating from scripting languages, such as Python to Julia. Practitioners typically use Julia for large-scale scientific data analysis, e.g., Julia was used in Celeste, a software used in astronomy research. The use of Julia yielded a performance improvement by a factor of 1,000 for Celeste, compared to prior implementation.   
```

Comment:
```suggestion
Since its inception in 2012, Julia has experienced a steady increase in popularity as practitioners are migrating from scripting languages, such as Python.
Practitioners typically use Julia for large-scale scientific data analysis.
For example, Julia was used in [Celeste](https://cs.lbl.gov/news-media/news/2016/celeste-enhancements-create-new-opportunities-in-sky-surveys/), a software used in astronomy research.
Julia yielded a 1000x performance improvement for [Celeste](https://arxiv.org/abs/1611.03404), compared to the prior implementation.   
```

Also, about that 1000x number...is that from changing to Julia alone or also from introducing distributed node parallelism? I don't know Julia...did distributed node parallelism just come for *free* with Julia? If not, it might be worth adding some clarification about that.

Reply:
Suggested change looks good to me.

The 1000 number is coming from a [blog](https://www.hpcwire.com/off-the-wire/julia-joins-petaflop-club/), which says: "Celeste achieved peak performance of 1.54 petaflops using 1.3 million threads on 9,300 Knights Landing (KNL) nodes of the Cori supercomputer at NERSC – a performance improvement of 1,000x in single-threaded execution." 

Will the phrase "The use of Julia on a supercomputer" make it sound better?

Reply:
Well, based on that article, it does indeed look like distributed node parallelism comes kinda sorta for free with Julia. So, I am fine leaving that 1000x statement as is.

Reply:
OK. Sounds good.